account_invoice_merge/invoice.py

```diff
@@ -200,17 +200,16 @@ def make_key(br, fields):
                             so_line.write(
                                 {'invoice_lines': [(6, 0, invoice_line_ids)]})
             if po_obj:
-                todo_ids = po_obj.search(
+                todos = po_obj.search(
                     [('invoice_ids', 'in', invoices_info[new_invoice_id])])
-                for org_po in po_obj.browse(todo_ids):
+                for org_po in todos:
```

Comment:
Remove for loop and do

    todos.write({'invoice_ids': [(4, new_invoice_id)]})